esrally/driver/runner.py

```diff
@@ -2068,6 +2071,62 @@ def __repr__(self, *args, **kwargs):
         return "delete-async-search"
 
 
+class OpenPointInTime(Runner):
+    async def __call__(self, es, params):
+        index = mandatory(params, "index", self)
+        response = es.open_point_in_time(index=index,
+                                         params=params)
+        id = response.get("id")
+        op_name = mandatory(params, "name", self)
+        # is name uniqueness enforced somewhere? this may collide with async searches
+        CompositeContext.put(op_name, id)
+
+    def __repr__(self, *args, **kwargs):
+        return "open-point-in-time"
+
+
+class QueryWithSearchAfterScrolling(Query):
+    async def __call__(self, es, params):
+        body = {"query": mandatory(params, "query", self),
+                "sort": mandatory(params, "sort", self)}
+        size = params.get("results-per-page", 10)
+        body["size"] = size
+        pit_op = params.get("with-point-in-time-from")
+        request_params, headers = self._transport_request_params(params)
+
+        while True:
+            if pit_op:
+                pit_id = CompositeContext.get(pit_op)
+                body["pit"] = {"id": pit_id,
+                               "keep_alive": "1m" }
+            props = await self._raw_search(es, doc_type=None, index=None, body=body.copy(), params=request_params, headers=headers)
```

Comment:
Is the `body.copy()` necessary here?

Reply:
interesting problem...

If you take it out, a test that asserts the correct order of a changing PIT key was used in this loop fails, due to (apparently) the `.._called_with` APIs themselves use objects as references rather than doing a deep copy.  The mutation in this loop causes the mock to ultimately assert that the _final_ iteration of the body is what was passed each time the ES client is exercised.

It is not functionally required, but I also desire to have the test in place that checks the thing is properly updated.  Thoughts?

Reply:
🤦. Makes sense now with your explanation and I fear you have already chosen the most pragmatic approach here. Maybe a comment is warranted here.

Reply:
Reading through this implementation, you seem to always paginate pages to exhaustion? I'm not sure entirely sure this is what we will always want to do when using this functionality.  Could we possibly limit the number of pages or the total number of results to paginate?